### ISSUE TITLE
atomic functions in statements

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#46]
Support for atomic functions is not yet fully backported to the master branch. It works in assignments:
```
gap>  f := atomic function(x) return x; end;
function( x ) ... end
```
but it doesn't work in statements:
```
gap>  atomic function(x) return x; end;
Syntax error: do or comma expected
 atomic function(x) return x; end;
                                 ^
```
We have to find the fix for this in HPC-GAP and properly backport it to the master branch.

**Comment by alex-konovalov** [comment]
@ChrisJefferson: In the hpcgap-default branch, this was achieved in commits https://github.com/gap-system/gap/commit/33ec3a58ba707093b812701d18988071508b5117 and https://github.com/gap-system/gap/commit/da2a27827c8eba9025cb779d8c05111a9122d27a

**Comment by ChrisJefferson** [comment]
Fixed in #53 (assuming tests pass!)

**Comment by ChrisJefferson** [comment]
merged and fixed
